models/demos/whisper/demo/demo.py

```diff
@@ -0,0 +1,543 @@
+        for i in range(batch_size):
+            s = next(sample)
+            inputs = feature_extractor(s["audio"]["array"], sampling_rate=16000, return_tensors="pt")
+            ground_truth_transcriptions = s["text"]
+            dtype_to_use = torch.bfloat16
+            input_features = inputs.input_features.type(dtype_to_use)
+
+            batched_inputs = input_features if i == 0 else torch.cat((batched_inputs, input_features), dim=0)
+
+            decoder_input_ids = torch.tensor([[1, 1]]) * config.decoder_start_token_id
+            decoder_input_ids = pad_input_32(decoder_input_ids, config.pad_token_id).to(torch.long)
+            batched_decoder_input_ids = (
+                decoder_input_ids if i == 0 else torch.cat((batched_decoder_input_ids, decoder_input_ids), dim=0)
+            )
+
+            batched_ground_truth_transcriptions.append(ground_truth_transcriptions)
+
+        parameters = preprocess_model_parameters(
+            initialize_model=lambda: model.eval(),
+            convert_to_ttnn=ttnn_model.convert_to_ttnn,
+            custom_preprocessor=ttnn_model.custom_preprocessor,
+            device=device,
+        )
+
+        (input_embeds, decoder_hidden_states, decoder_attention_mask) = ttnn_model.preprocess_inputs(
+            config=config,
+            input_features=batched_inputs,
+            input_ids=batched_decoder_input_ids,
+            attention_mask=None,
+            parameters=parameters,
+            device=device,
+            whisper_memory_config=whisper_memory_config,
+        )
+
+        ttnn_output = run_generate(
+            config,
+            input_embeds,
+            batched_inputs,
+            ttnn_model,
+            decoder_hidden_states,
+            decoder_attention_mask=decoder_attention_mask,
+            parameters=parameters,
+            ttnn_linear_weight=ttnn_linear_weight,
+            device=device,
+            decoder_input_ids=batched_decoder_input_ids,
+            generation_config=hf_reference_model.generation_config,
+            batch_size=batch_size,
+            max_tokens=max_tokens,
+            whisper_memory_config=whisper_memory_config,
+        )
+        ttnn_transcription = processor.batch_decode(ttnn_output, skip_special_tokens=True)
+
+        logger.info("Model Output")
+        logger.info(ttnn_transcription)
+
+        wer = WordErrorRate()
+        wer_scores = []
+        for transcription, ground_truth in zip(ttnn_transcription, batched_ground_truth_transcriptions):
+            transcription = transcription.upper()
+            individual_wer_score = wer([transcription], [ground_truth])
+            wer_scores.append(individual_wer_score)
+            logger.info(f"Individual Sample WER score: {individual_wer_score}")
+
+        average_wer_score = sum(wer_scores) / len(wer_scores)
+        logger.info(f"Average WER score: {average_wer_score}")
+        accuracy = 1 - average_wer_score
+        logger.info(f"Accuracy: {accuracy}")
```

Comment:
Is there an expected accuracy that should be asserted against? (same comment for other demo tests)

Reply:
I've updated the demo to include an accuracy assertion with a 3% tolerance.